javascript/readme.md

```diff
@@ -1,3 +1,386 @@
+
+```javascript
+function outer(c, d) {
+    var e = c * d;
+
+    function inner(a, b) {
+        return (e * a) + b;
+    }
+
+    return inner(0, 1);
+}
+```
+
+This convention works well with JavaScript because in JavaScript, functions and object literals can be placed anywhere that an expression is allowed. It provides the best readability with inline functions and complex structures.
+
+```javascript
+function getElementsByClassName(className) {
+    var results = [];
+    walkTheDOM(document.body, function (node) {
+        var array;                // array of class names
+        var ncn = node.className; // the node's classname
+
+		// If the node has a class name, then split it into a list of simple names.
+		// If any of them match the requested name, then append the node to the list of results.
+
+        if (ncn && ncn.split(" ").indexOf(className) >= 0) {
+            results.push(node);
+        }
+    });
+    return results;
+}
+```
+
+If a function literal is anonymous, there should be one space between the word function and the ( left parenthesis. If the space is omitted, then it can appear that the function's name is function, which is an incorrect reading.
+
+```javascript
+div.onclick = function (e) {
+    return false;
+};
+
+that = {
+    method: function () {
+        return this.datum;
+    },
+    datum: 0
+};
+```
+
+Use of global functions should be minimized.
+
+When a function is to be invoked immediately, the entire invocation expression should be wrapped in parens so that it is clear that the value being produced is the result of the function and not the function itself.
+
+```javascript
+var collection = (function () {
+    var keys = [];
+    var values = [];
+
+    return {
+        get: function (key) {
+            var at = keys.indexOf(key);
+            if (at >= 0) {
+                return values[at];
+            }
+        },
+        set: function (key, value) {
+            var at = keys.indexOf(key);
+            if (at < 0) {
+                at = keys.length;
+            }
+            keys[at] = key;
+            values[at] = value;
+        },
+        remove: function (key) {
+            var at = keys.indexOf(key);
+            if (at >= 0) {
+                keys.splice(at, 1);
+                values.splice(at, 1);
+            }
+        }
+    };
+}());
+```
+
+## Names
+
+Names should be formed from the 26 upper and lower case letters (A .. Z, a .. z), the 10 digits (0 .. 9), and _ underbar. Avoid use of international characters because they may not read well or be understood everywhere. Do not use `$` dollar sign or `\` backslash in names.
```

Comment:
Should we maybe settle on a format for naming, such as camelCaseFunction()? Also we should make sure that functions have descriptive names.
initHomepageSlider() & isCheckoutValid()